data_model/src/asset.rs

```diff
@@ -585,20 +661,6 @@ impl DefinitionId {
     }
 }
 
-impl Id {
-    /// Construct [`Id`] from [`DefinitionId`] and [`AccountId`].
-    #[inline]
-    pub const fn new(
-        definition_id: <AssetDefinition as Identifiable>::Id,
-        account_id: <Account as Identifiable>::Id,
-    ) -> Self {
-        Self {
-            definition_id,
-            account_id,
-        }
-    }
-}
```

Comment:
please don't move this around because you'll end up in conflict with someone else. There is a lot of parallel work on `data_model`